projects/dendrites/permutedMNIST/analyze_results.py

```diff
@@ -72,47 +107,67 @@ def parse_one_experiment(exp, state, df):
                     results = trial_checkpoint["results"]
                     if results is None:
                         continue
-
-                    # For each checkpoint select the iteration with the best accuracy as
-                    # the best epoch
-                    best_results = max(results,
-                                       key=lambda x: x.get("mean_accuracy", 0.0))
-                    best_result = best_results["mean_accuracy"]
-                    if best_result > 0.0:
-                        # Get the trial parameters we care about
-                        config = trial_checkpoint["config"]
-                        model_args = config["model_args"]
-                        kw_percent_on = model_args["kw_percent_on"]
-                        weight_sparsity = model_args.get("weight_sparsity", 0.0)
-                        dendrite_weight_sparsity = model_args.get(
-                            "dendrite_weight_sparsity", 0.0)
-                        num_segments = model_args.get("num_segments")
-                        dim_context = model_args["dim_context"]
-                        epochs = config["epochs"]
-                        num_tasks = config["num_tasks"]
-                        lr = config["optimizer_args"]["lr"]
-                        momentum = config["optimizer_args"].get("momentum", 0.0)
-
-                        # This list must match the column headers in collect_results
-                        df_entries.append([
-                            exp, kw_percent_on, weight_sparsity,
-                            dendrite_weight_sparsity, num_segments, dim_context,
-                            epochs, num_tasks, lr, momentum,
-                            config["seed"], best_result,
-                            "{} {}".format(exp, tag)
-                        ])
+                    if outmethod == "best":
+                        # For each checkpoint select the iteration with the best accuracy as
+                        # the best epoch
+                        print("using parsing method : best")
+                        best_results = max(
+                            results, key=lambda x: x.get("mean_accuracy", 0.0)
+                        )
+                        best_result = best_results["mean_accuracy"]
+                        if best_result > 0.0:
+                            parse(
+                                best_result,
+                                results,
+                                trial_checkpoint,
```

Comment:
For better aesthetics, I recommend not having 1 argument per line, but instead all on the same line and wrapping, e.g.

```
if best_result > 0.0:
    parse(argument_1, argument_2, argument_3,
          argument_4, argument_5)
```

Reply:
The above comment applies to 3 different calls to `parse`